src/controllers/User.ts

```diff
@@ -163,9 +164,36 @@ export async function destroy(req: Request, res: Response) {
     } catch (e) {
         console.error(e);
         if ((e as Error).name == "not found") res.status(404).json({ message: (e as Error).message });
-        else res.status(500).json({ message: "An error occurred" });
+        else res.status(500).json(errorJson("Error Occurred", null));
     } finally {
         client.release();
     }
 }
 
+
+
+// Hash function
+export async function hashPassword(password: string): Promise<string> {
+    const saltRounds = 10;
+    return await bcrypt.hash(password, saltRounds);
+}
+
+
+// store image
+export async function storeImage(profile: UploadedFile): Promise<string> {
+    return new Promise((resolve, reject) => {
+        const fileName = Date.now().toString() + "-" + profile.name;
+        const uploadPath = path.join(__dirname, "../../public", fileName);
+
+        profile.mv(uploadPath, (err) => {
+            if (err) {
+                console.error("Error moving file:", err);
+                reject(err);
+            } else {
+                resolve(fileName);
+            }
+        });
+    });
+}
```

Comment:
It would be better if error handling is added in this function. 
example: 
```typescript
// store image
export async function storeImage(profile: UploadedFile): Promise<string> {
    try {
        const fileName = Date.now().toString() + "-" + profile.name;
        const uploadPath = path.join(__dirname, "../../public", fileName);

        await profile.mv(uploadPath);
        return fileName;
    } catch (err) {
        if (err.code === 'EACCES') {
            console.error('Error storing image: permission denied');
            throw new Error('Permission denied while storing image');
        } else if (err.code === 'ENOSPC') {
            console.error('Error storing image: no space left on device');
            throw new Error('No space left on device while storing image');
        } else {
            console.error('Error storing image:', err);
            throw new Error('Error storing image');
        }
    }
}
```